endpoints/openrtb2/auction.go

```diff
@@ -1154,7 +1160,11 @@ func checkSafari(r *http.Request) (isSafari bool) {
 // Write(return) errors to the client, if any. Returns true if errors were found.
 func writeError(errs []error, w http.ResponseWriter) bool {
 	if len(errs) > 0 {
-		w.WriteHeader(http.StatusBadRequest)
+		if errortypes.DecodeError(errs[0]) == errortypes.BlacklistedAppCode {
```

Comment:
I understand that the way the code is arranged right now, if `BlacklistedApp` error occurs, it will be the first one in the `errs` slice but if we make any changes to the code in the future then this might or might not be true. And this will also be difficult to catch at the time. For example, right now this is true because you re-arranged the imp validation block. If the imp validation block were still where it was before, the `BlacklistedApp` error might not be the first one in the `errs` slice. And so, I'd suggest we rather go through the slice and check all the errors than just the first one. Maybe we can add the `BlacklistedApp` error as a check in the `fatalError` function?